libs/generic-view/ui/src/lib/buttons/buttons.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import ButtonText from "./button-text"
+import ButtonLink from "./button-link"
+
+export const buttons = {
```

Comment:
⚪ Neutral, Suggestion:

Continuing the discussion from the previous section https://github.com/mudita/mudita-center/pull/1667/files#r1457039056

```tsx
/**
 * The `Button` component handles various variants, such as text, link, icon, or even a custom button.
 * This serves as a compromise between a generic button component and a more complex parent component that manages navigation and modal usage.
 * Variants allow flexible customization of the button's appearance and functionality for specific use cases while preserving overall control in the parent component.
 */
interface ButtonProps extends BaseButtonProps {
  type: 'default' | 'primary' | 'secondary';
  theme: Theme;
  variant: 'text' | 'link' | 'custom'; // Add new variants
  action: ButtonAction;
}

const Button: FunctionComponent<ButtonProps> = ({
  theme,
  type,
  variant,
  action,
  ...restProps
}) => {
  // ... (Remainder of the component unchanged)

  // Depending on the variant, render the appropriate component
  switch (variant) {
    case 'text':
      return <ButtonText {...props} onClick={handleButtonClick} />;
    case 'link':
      return <ButtonLink {...props} onClick={handleButtonClick} />;
    case 'icon':
      return <BaseButtonWithIcon {...props} onClick={handleButtonClick} />;
    case 'custom':
      // Add handling for a custom variant if necessary
      return <CustomButton {...props} onClick={handleButtonClick} />;
    default:
      return <BaseButton {...props} onClick={handleButtonClick} />;
  }
};
```
This component has been adapted to accommodate diverse variants, offering a balance between a generic button component and a more intricate parent component responsible for navigation and modal usage. Variants enable adaptable customization of the button's visual and functional aspects for specific use cases, all while maintaining overarching control in the parent component.